editor/components/post-taxonomies/flat-term-selector.js

```diff
@@ -105,26 +105,24 @@ class FlatTermSelector extends Component {
 	}
 
 	findOrCreateTerm( termName ) {
-		return new Promise( ( resolve, reject ) => {
-			// Tries to create a term or fetch it if it already exists
-			const basePath = wp.api.getTaxonomyRoute( this.props.slug );
-			apiRequest( {
-				path: `/wp/v2/${ basePath }`,
-				method: 'POST',
-				data: { name: termName },
-			} ).then( resolve, ( xhr ) => {
-				const errorCode = xhr.responseJSON && xhr.responseJSON.code;
-				if ( errorCode === 'term_exists' ) {
-					// search the new category created since last fetch
-					this.addRequest = apiRequest( {
-						path: `/wp/v2/${ basePath }?${ stringify( { ...DEFAULT_QUERY, search: termName } ) }`,
-					} );
-					return this.addRequest.then( ( searchResult ) => {
-						resolve( find( searchResult, ( result ) => isSameTermName( result.name, termName ) ) );
-					}, reject );
-				}
-				reject( xhr );
-			} );
+		const basePath = wp.api.getTaxonomyRoute( this.props.slug );
+		// Tries to create a term or fetch it if it already exists
+		return apiFetch( {
+			path: `/wp/v2/${ basePath }`,
+			method: 'POST',
+			data: { name: termName },
+		} ).catch( ( error ) => {
+			const errorCode = error.code;
+			if ( errorCode === 'term_exists' ) {
+				// search the new category created since last fetch
+				this.addRequest = apiFetch( {
+					path: `/wp/v2/${ basePath }?${ stringify( { ...DEFAULT_QUERY, search: termName } ) }`,
+				} );
+				return this.addRequest.then( ( searchResult ) => {
```

Comment:
I find that using `async/await` is not necessarily better when we use `.catch` because you end up with `await` inside try/catch and you also have to embed `try/catch` inside `try/catch` sometimes.